src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -0,0 +1,511 @@
+package au.org.emii.gogoduck.worker;
+
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.opengis.util.ProgressListener;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+
+public class GoGoDuck {
```

Comment:
I think the `GoGoDuck` bit is redundant, since anyone using this class would already be working in that context.  It's ok if you like it though!